x/incentives/keeper/distribute.go

```diff
@@ -602,6 +602,27 @@ func (k Keeper) distributeInternal(
 		// Get distribution epoch duration. This is used to calculate the emission rate.
 		currentEpoch := k.GetEpochInfo(ctx)
 
+		// Validate that the gauge's corresponding uptime is authorized.
```

Comment:
I would personally move all this new logic into a new function `k.getNoLockGaugeUptime` to manage complexity better.

Reply:
Thanks for calling this out. Addressed in a91496028ad25f8b74c74ae0a83cc38cd598f852